### PR TITLE
Wagtail 2.10.1

### LINKED ISSUE #5065: Wagtail 2.10 upgrade

### PR DESCRIPTION
**Pull request by Pomax**
Closes #5065 

- updates Wagtail to 2.10.1
- also updated Django to 2.2.15
- has a shimmed `ImageFactory` so that images are always generated into the `Root` collection instead of their own collection
- still fails `inv test-python` on a site redirect test:

```
======================================================================
FAIL: test_PNI_homepage_redirect_to_foundation_site (network-api.networkapi.tests.RedirectDefaultSiteDecoratorTests)
Test that users gets redirected to PNI on the foundation site when they visit it from a non-default CMS site
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/network-api/networkapi/tests.py", line 141, in test_PNI_homepage_redirect_to_foundation_site
    fetch_redirect_response=False
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/test/testcases.py", line 383, in assertRedirects
    msg_prefix + "Response redirected to '%s', expected '%s'" % (url, expected_url)
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/test/testcases.py", line 403, in assertURLEqual
    msg_prefix + "Expected '%s' to equal '%s'." % (url1, url2)
AssertionError: 'https://localhost/en/privacynotincluded/' != 'https://default-site.com/en/privacynotincluded/'
- https://localhost/en/privacynotincluded/
?           ^^^^^^^
+ https://default-site.com/en/privacynotincluded/
?         +++++ ++++++++ ^
 : Response redirected to 'https://localhost/en/privacynotincluded/', expected 'https://default-site.com/en/privacynotincluded/'Expected 'https://localhost/en/privacynotincluded/' to equal 'https://default-site.com/en/privacynotincluded/'.
```

Error occurs in the https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/tests.py#L111-L142 batch, specifically the https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/tests.py#L138-L142 test, and it feels like the Site declarations on https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/tests.py#L112-L115 no longer work.

### CODE REVIEW

File: network-api/networkapi/settings.py [hunk 1]
```diff
@@ -249,7 +249,6 @@
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
 
-    'wagtail.core.middleware.SiteMiddleware',
```
Comment (Pomax) [inline network-api/networkapi/settings.py:252]: as per Wagtail 2.9, see https://docs.wagtail.io/en/stable/releases/2.9.html#sitemiddleware-and-request-site-deprecated

File: network-api/networkapi/wagtailpages/factory/image_factory.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import factory
+from wagtail.images import get_image_model
+from wagtail.core.models import Collection
+
+
+# Slightly modify the wagtail_factories ImageFactory so that it
+# always generates images in the Root collection:
+
+class CollectionMemberFactory(factory.DjangoModelFactory):
+    collection = Collection.objects.get(name='Root')
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/factory/image_factory.py:10]: wagtail_factories uses `collection = factory.SubFactory(CollectionFactory, parent=None)` here

File: network-api/networkapi/wagtailpages/views.py [hunk 3]
```diff
@@ -17,7 +19,7 @@ def custom404_view(request, exception):
     belongs to which site, as "a site" is not tied to "a django
     app" in the wagtail way of things.
     """
-    if request.site.hostname == 'www.mozillafestival.org':
+    if Site.find_for_request(request).hostname == 'www.mozillafestival.org':
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/views.py:22]: as per Wagtail 2.9, see https://docs.wagtail.io/en/stable/releases/2.9.html#sitemiddleware-and-request-site-deprecated

File: network-api/networkapi/settings.py (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/settings.py:252]: Yep, good catch!

File: network-api/networkapi/wagtailpages/factory/image_factory.py (hunk 2 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/factory/image_factory.py:10]: I dont see any more collections in localhost. Nicely done!

File: network-api/networkapi/tests.py [hunk 4]
```diff
@@ -109,8 +110,14 @@ def test_mozilla_foundation_users_not_deleted(self):
 
 
 class RedirectDefaultSiteDecoratorTests(TestCase):
+
     def setUp(self):
         self.factory = RequestFactory()
+        # Change the default site away from localhost
+        self.original_default_site = Site.objects.get(is_default_site=True, hostname='localhost')
+        self.original_default_site.is_default_site = False
```
Comment (KalobTaulien) [inline network-api/networkapi/tests.py:118]: I added this to "remove" the default site. Originally, `localhost` is the default site. These few lines unset `localhost` as the default site so we can add a custom one.

File: network-api/networkapi/tests.py [hunk 5]
```diff
@@ -141,6 +148,15 @@ def test_PNI_homepage_redirect_to_foundation_site(self):
...
         )
 
+    def tearDown(self):
+        # Re-instante localhost as the default site
+        self.original_default_site.is_default_site = True
+        self.original_default_site.save()
+
+        # Remove the Site Factories
+        self.default_site.delete()
+        self.secondary_site.delete()
```
Comment (KalobTaulien) [inline network-api/networkapi/tests.py:158]: And when we're done with this class we'll re-instate `localhost` as the "default" site, and clean up the other sites we generated.

File: requirements.in [hunk 6]
```diff
@@ -1,6 +1,6 @@
 beautifulsoup4
 boto3
-Django==2.2.14
+Django==2.2.15
```
Comment (KalobTaulien) [inline requirements.in:3]: Why not jump to Django 3, do we have dependencies that aren't updated yet?

Comment (Pomax) [reply requirements.in:3]: one uplift at a time.

Comment (cadecairos) [inline requirements.in:3]: let's make this 2.2.16 (security update)

Comment (Pomax) [reply requirements.in:3]:
```suggestion
Django==2.2.16
```

**Comment by cadecairos** [comment]
@Pomax is this still a draft PR?

**Comment by Pomax** [comment]
not anymore, I had to bump the django version and rerun the tests, but it's all good now and ready for "real" review =D

**Review by cadecairos — APPROVED** [review]
R+

Tested with a staging data copy too, things looked great locally.

**Comment by cadecairos** [comment]
I think a rebase help with the percy failure?

**Comment by Pomax** [comment]
rebase done, but the percy failure is something that's -annoyingly- not fully fixed, seeing it on other PRs as well.

**Review by KalobTaulien — APPROVED** [review]
This is working. Good to merge in my opinion once percy passes.

**Comment by cadecairos** [comment]
I'm okay with ignoring the Percy fail if we file a new issue to address the lingering random generation problems.

**Comment by Pomax** [comment]
already filed and assigned: https://github.com/mozilla/foundation.mozilla.org/issues/5202

**Pull request merged**